RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -56,8 +56,11 @@ void HGCalCLUEAlgo::populate(const HGCRecHitCollection& hits) {
     int offset = ((rhtools_.zside(detid) + 1) >> 1) * maxlayer_;
     int layer = layerOnSide + offset;
     cells_[layer].detid.emplace_back(detid);
+    cells_[layer].isSilic.emplace_back(rhtools_.isSilicon(detid));
     cells_[layer].x.emplace_back(position.x());
     cells_[layer].y.emplace_back(position.y());
+    cells_[layer].eta.emplace_back(position.eta());
+    cells_[layer].phi.emplace_back(position.phi());
```

Comment:
My understanding is that the process of constructing the data structure is not CPU intensive (as slide 19 of https://indico.cern.ch/event/817359/contributions/3420465/attachments/1839025/3014182/modifiedClustering.pdf). @rovere @felicepantaleo Please correct me if I am wrong. If this is a concern, I can try to fix this but this might need a bigger surgery to the code since we got layers mixed with silicon and scintillator.